packages/@webex/plugin-cc/src/AgentConfigService/AgentConfigService.ts

```diff
@@ -0,0 +1,126 @@
+import {WebexSDK, HTTP_METHODS} from '../types';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from './types';
+import Request from '../request';
+
+export default class AgentConfigService {
+  ciUserId: string;
+  orgId: string;
+  webex: WebexSDK;
+  wccAPIURL: string;
+  requestInstance: Request;
+
+  constructor(ciUserId: string, orgId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.ciUserId = ciUserId;
+    this.orgId = orgId;
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+    this.requestInstance = new Request(this.webex);
+  }
+
+  /**
+   * Method to get User using CI.
+   * @returns {Promise<UserResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getUserUsingCI(): Promise<UserResponse> {
+    try {
+      const URL = `${this.wccAPIURL}organization/${this.orgId}/user/by-ci-user-id/${this.ciUserId}`;
+      const response = await this.requestInstance.request(URL, HTTP_METHODS.GET);
+
+      this.webex.logger.log('getUserUsingCI api called successfully.');
+
+      return Promise.resolve(response?.body);
+    } catch (error) {
+      return Promise.reject(new Error(`Error while calling getUserUsingCI API, ${error}`));
+    }
+  }
+
+  /**
+   * Method to get Desktop Profile by passing Org Id.
+   * @param {string} desktopProfileId ID of the Desktop Profile to be retrieved.
+   * @returns {Promise<DesktopProfileResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getDesktopProfileById(desktopProfileId: string): Promise<DesktopProfileResponse> {
+    try {
+      const URL = `${this.wccAPIURL}organization/${this.orgId}/agent-profile/${desktopProfileId}`;
+      const response = await this.requestInstance.request(URL, HTTP_METHODS.GET);
+      this.webex.logger.log('retrieveDesktopProfileById api called successfully.');
+
+      return Promise.resolve(response?.body);
+    } catch (error) {
+      return Promise.reject(
+        new Error(`Error while calling retrieveDesktopProfileById API, ${error}`)
+      );
+    }
+  }
+
+  /**
+   * Method to get List of Teams.
+   * @param {number} page Index of the page of results to be fetched.
+   * @param {number} pageSize Number of items to be displayed on a page.
+   * @param {Array<String>} filter Filter which can be applied to the elements to be fetched.
+   * @param {Array<String>} attributes Specify the attributes to be returned.
+   * @returns {Promise<ListTeamsResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getListOfTeams(
+    page?: number,
```

Comment:
Except for the filter here, other parameters here are defined constants. Why are they optional? Same comment applicable for next method as well

Reply:
Code Updated!